aptos-move/framework/aptos-framework/sources/storage_gas.move

```diff
@@ -101,10 +104,16 @@ module aptos_framework::storage_gas {
     }
 
     public fun new_point(x: u64, y: u64): Point {
+        assert!(
+            x <= BASIS_POINT_DENOMINATION && y <= BASIS_POINT_DENOMINATION,
+            error::invalid_argument(EINVALID_POINT_RANGE)
+        );
         Point {x, y}
     }
 
     public fun new_gas_curve(min_gas: u64, max_gas: u64, points: vector<Point>): GasCurve {
+        assert!(max_gas >= min_gas, error::invalid_argument(EINVALID_GAS_RANGE));
+        assert!(max_gas <= MAX_U64 / BASIS_POINT_DENOMINATION, error::invalid_argument(EINVALID_GAS_RANGE));
```

Comment:
Slight rounding error with MAX_U64 / BASIS_POINT_DENOMINATION but that's fine as it keeps max_gas more conservative I guess.

Reply:
Yes, I think so too. It's rounding down, and should make `max_gas` more conservative.

Reply:
These were checked in `validate_curve()`?

Reply:
Yeah, right. They seem redundant. These are different from the ones in `validate_curve()` in a subtle way. Here, these are checked before constructing a new `GasCurve`, so it make the following `struct` invariants provable:
```
    spec GasCurve {
        invariant min_gas <= max_gas;
        invariant max_gas <= MAX_U64 / BASIS_POINT_DENOMINATION;
    }
```

Reply:
To avoid these redundant checks, it may be ideal to unify the two different styles (checking (1) before and (2) after constructing structs) in the module. But, this would require further refactoring...

Reply:
My understanding is if we check when constructing and the curve is immutable, we do not have to validate after that. We could safety delete `validate_curve`, right？

Reply:
Yes!! `curve`s can be validated "by construction". You're smart! :)
We can safety delete `validate_curve` if we also validate the `points` at the time of construction. I will make the change now.

Reply:
@lightmark, I removed `validate_curve` and updated the code accordingly. Please let me know what you think.

One who mutates `curve` inside of the module have to make sure that the new value holds the validation condition. Otherwise, Prover will complain about the broken struct invariant in CI (once Prover becomes a land blocker in the future).

Reply:
n00b: How does the prover check runtime modification if the module opens a door for that?

Reply:
As you know, by default, the fields of GasCurve cannot be directly modified outside of the module.

If this module provide a “setter” function, the modification actually happens within the module, and Prover will check if the setter function properly validates the input.

If you allows the module to export a mut ref of a field of the struct, then it can be modified by an unknown module at runtime. Prover cannot help, unfortunately. However, I believe this is an anti pattern because it breaks the modularity. We rarely use this pattern. One exception is the vector module.